pnp/PnpHelper.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+
+namespace PnpHelpers
+{
+    public static class PnpHelper
+    {
+        public const string TelemetryComponentPropertyName = "$.sub";
+        private const string PropertyComponentIdentifier = "\"__t\": \"c\",";
+
+        public static string CreateReportedPropertiesPatch(string propertyName, object propertyValue, string componentName = null)
+        {
+            if (string.IsNullOrWhiteSpace(componentName))
+            {
+                return $"" +
+                    $"{{" +
+                    $"  \"{propertyName}\": " +
+                    $"      {{ " +
+                    $"          \"value\": \"{propertyValue}\" " +
+                    $"      }} " +
+                    $"}}";
+            }
+
+            return $"" +
+                $"{{" +
```

Comment:
Just a thought experiment, how would this work for nested component cases?

Reply:
good point, do we have a sample patch for that? All the samples in the spec doc were single component examples.
If I can get a nested component patch, I can see how to make this more flexible.

Reply:
No nested componenets - there is only one level of component supported.